addons/hr_holidays/models/hr_leave_accrual_plan_level.py

```diff
@@ -5,25 +5,19 @@
 import calendar
 
 from dateutil.relativedelta import relativedelta
-from num2words import num2words
 
 from odoo import _, api, fields, models
 from odoo.tools.date_utils import get_timedelta
-from odoo.tools.misc import get_lang
 
 
 DAYS = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat']
 MONTHS = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec']
 # Used for displaying the days and reversing selection -> integer
 DAY_SELECT_VALUES = [str(i) for i in range(1, 29)] + ['last']
+DAY_SELECT_SELECTION_NO_LAST = tuple(zip(DAY_SELECT_VALUES, (str(i) for i in range(1, 29))))
 
 def _get_selection_days(self):
-    lang = get_lang(self.env).code
-    return [
-        (DAY_SELECT_VALUES[i - 1],
-        num2words(i, lang=lang, to='ordinal_num') if i < 29 else _('last day'))
-        for i in range(1, 30)
-    ]
```

Comment:
I'd keep `num2words` here but fallback to English if the language is not supported [1]

[1] : https://github.com/savoirfairelinux/num2words#:~:text=except%20NotImplementedError%3A

Reply:
After talking with gmf, we agree it's better to use the number bcs making something grammatically correct in all language is too complex, (aka 2nd of July in english, 2 juillet in french) so using number is easier and more universal.
On top of that, in the xml the use of XXX_day and XXX_day_display seem to be used interchangeably (or at least in similar situation)

Reply:
Ok for me if GMF agrees then :-)